src/main/java/com/volasoftware/tinder/service/UserServiceImpl.java

```diff
@@ -188,4 +190,30 @@ public void removeFriend(Long friendId) {
         user.getFriends().remove(friend);
         userRepository.save(user);
     }
+
+    public List<User> getUsersByUserType(UserType userType) {
+        return userRepository.findByType(userType);
+    }
+
+    public List<User> linkRandomFriendsForNonBotUsers() {
+        List<User> nonBotUsers = userRepository.findByType(UserType.REAL);
+        List<User> botUsers = userRepository.findByType(UserType.BOT);
+
+        FriendLinker.linkRandomFriendsForNonBotUsers(nonBotUsers, botUsers);
+
+        return nonBotUsers;
+    }
+
+    public User linkRandomFriendsForRequestedUser(Long userId) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        User requestedUser = optionalUser.orElseThrow(() ->
+                new UserDoesNotExistException("User with this id does not exist"));
```

Comment:
This can be refactored to a single row
```suggestion
        User requestedUser = userRepository.findById(userId).orElseThrow(() ->
                new UserDoesNotExistException("User with this id does not exist"));
```